### PR TITLE
Routine:Add RoutinAgent

### PR DESCRIPTION
**Pull request by kimhyungrok**
It add the RoutineAgent and related code for handling routine interface.

It update the CapabilityFactory for providing RoutineAgent

It add to construct the RoutineAgent in nugu_sample.					

Signed-off-by: Hyungrok.Kim <hr97gdi@sk.com>

### CODE REVIEW

**Review by JeanTracker — APPROVED** [review]

File: src/capability/routine_agent.cc [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+}
+
+std::string RoutineAgent::sendEventActionTriggered(const std::string& ps_id, const Json::Value& data, EventResultCallback cb)
+{
+    if (ps_id.empty() || data.empty()) {
+        nugu_error("The mandatory data for sending event is missed.");
+        return "";
+    }
+
+    Json::StyledWriter writer;
```
Comment (webispy) [inline src/capability/routine_agent.cc:283]: Please use `FastWriter` instead of `StyledWriter`

File: src/capability/routine_agent.cc [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+}
+
+void RoutineAgent::sendEventCommon(std::string&& event_name, Json::Value&& extra_value, EventResultCallback cb, bool clear_routine_info)
+{
+    if (play_service_id.empty()) {
+        nugu_error("The mandatory data for sending event is missed.");
+        return;
+    }
+
+    Json::StyledWriter writer;
```
Comment (webispy) [inline src/capability/routine_agent.cc:299]: Please use `FastWriter` instead of `StyledWriter`

File: src/capability/routine_agent.cc [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+void RoutineAgent::deInitialize()
+{
+    if (timer) {
+        timer->stop();
+        timer.reset();
+    }
+
+    routine_manager->removeListener(this);
+
```
Comment (webispy) [inline src/capability/routine_agent.cc:68]: `setDataRequester(...)` is called in `initialize()` line 49, but there seems to be no cancellation logic for this. (maybe `routine_manager->reset()` ?)

File: src/capability/routine_agent.cc (hunk 2 above)
Comment (kimhyungrok) [reply src/capability/routine_agent.cc:299]: OK. I'll fix it.

File: src/capability/routine_agent.cc (hunk 1 above)
Comment (kimhyungrok) [reply src/capability/routine_agent.cc:283]: OK. I'll fix it.

File: src/capability/routine_agent.cc (hunk 3 above)
Comment (kimhyungrok) [reply src/capability/routine_agent.cc:68]: When deInitialize() of each CapabilityAgent is called as the CapabilityManager call reset() of all of the manager (include RoutineManager), it's no problem.

Comment (webispy) [reply src/capability/routine_agent.cc:68]: Ok. Thank you :)

**Review by webispy — APPROVED** [review]

**Pull request merged**